### PR TITLE
This branch is to refactor classes in OOP School Library project

### PR DESCRIPTION
**Pull request by steveWDamesJr**
## Refactor Classes

:sunglasses: This is @steveWDamesJr 's  and @levy002 's 1st group milestone for the OOP-School-Library build. :sunglasses:

- [x] Ensure that YAGNI, KISS, and DRY rules and Ruby style guide are respected
- [x] Follow SOLID principles.
- [x] Adhere to "Single-responsiblity Principle":
        - Ensure that the role of `main.rb` is only to start our code and ask the user for the initial  choice of options (or to quit).
        - Extract some pieces of logic into separate class(es).

- [x] Debug linters

   
    

Refactor

Thank you for taking the time to review our PR :pray:

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -15,87 +18,44 @@ def initialize
...
-    name = Services.name_input
-    specialization = Services.specialization_input
-    [age, name, specialization]
-  end
-
   def create_person
     print "\nDo you want to create a student (1) or a teacher (2)?[Input the number]: "
     @user_input = gets.chomp.to_s
     create_person unless users_input_valid?(user_input, %w[1 2])
     if @user_input == '1'
```
Comment (FrederickMih) [inline app.rb]: - [x] Hi team, you guys are doing so great on this project :+1: . All requests for inputs and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input hence this is a bad example of a single responsibility rule. Let's kindly refactor this again.
- [Best Practices lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid)

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



# To Highlights

- PR has a title and good summary :heavy_check_mark: 
- GitHub workflow has been followed :heavy_check_mark: 
- Linters checks are passing :heavy_check_mark: 
- README file is descriptive :heavy_check_mark: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (levy002) [reply app.rb]: Thank you @FrederickMih for the excellent suggestions. We've implemented it.

**Review by Temmarie — APPROVED** [review]
# Status: Approved ✔️ 
## Hi @steveWDamesJr  and @levy002  , 
### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 

#### Cheers and Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
------
_As described in the [Code reviews limits policy](<https://microverse.zendesk.com/hc/en-us/articles/1500004088561>) you have a limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](<https://airtable.com/shrQAqnBwek5a0O0s>)._

**Pull request merged**